weave/trace/__init__.py

```diff
@@ -0,0 +1,274 @@
+"""The top-level functions for Weave Trace API."""
```

Comment:
I'm not sure I understand the benefit of having this in `__init__.py` in general i try to stay away from populating `init.py` for a few reasons:
1. `__init__.py` are automatically executed when doing things like `from weave.trace.X import Y`
2. It makes it less clear what the purpose is of the file.

Can you share why you think `api.py` should be replaces with this?

Reply:
I added `api.py` as a stop-gap to help split out the `query` and `trace` apis.  Now that this is just trace, we don't need it.

The pattern I want is `from weave.trace import *` gives you all the symbols you would need in 90% of use cases.

I agree that having definitions in `__init__.py` is bad form.  In general `__init__` should just be imports, not defining anything.  I considered pulling out the definitions in this PR but it seemed noisier (we can pull symbols out in another PR)

If you prefer, I can pull the symbols out in this PR

Reply:
Hmmm, i disagree with `from weave.trace import *`. I would never do that in my own code. IMO, `import *` is not a clean pattern we should be promoting.

Even so, i would prefer `__init__.py` to only be used for explicitly re-exporting symbols or doing module-level initialization you want to run on each import of sub modules. I am not seeing the advantage here.

Reply:
In addition, we should not be working towards `weave.trace` for users anyway. They should not even know about trace server, so all of weave is "trace"

Reply:
I'd like to see some example of why `import *` is better. I only see less characters typed. Other than that, it feels like there are many footguns and you much with the global namespace.

Reply:
I think we are prematurely optimizing here. `flow` was an artifact of `weaveflow` and not something users should ever think about. I don't see how this adds user value, but in exchange we have more circular imports and implicit symbols being imported and exported